src/provider/apiresponse.h

```diff
@@ -0,0 +1,54 @@
+#ifndef ERD_PROXY_API_RESPONSE_H
+#define ERD_PROXY_API_RESPONSE_H
+
+#include "json/json.hpp"
+#include "../utils/errors.h"
+
+class ErdGenericApiResponse
+{
+public:
+    explicit ErdGenericApiResponse(std::string rawData)
```

Comment:
Usually, for this kind of object construction, we define a static function called `parse...()` that returns the object. Just a suggestion - fine as it is, though.

Reply:
I've applied your suggestion and created a function that parses `wrapper::http::Result` and returns an `ErdGenericApiResponse`:
`ErdGenericApiResponse parse(wrapper::http::Result const &res)`